### ISSUE TITLE
wgengine/magicsock: assume home DERP of exit node when in use

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#4643]
As a battery/bandwidth optimization, especially for mobile:

When using an exit node, we should just assume the exit node's preferred DERP server, changing our home if/when the exit node's changes.

That potentially reduces the number of DERP connections by one (along with its keep-alive heartbeats) and lets us STUN a bit less.  (don't need STUN to measure a home, only to figure out connection endpoints/properties)

If we do that, we might want to send a bool to in Hostinfo that `DERPHomeIsThatOfExitNode bool` so stats/debugging aren't confused by seeing a wrong-looking reported DERP home.

/cc @maisem @crawshaw

**Comment by danderson** [comment]
I don't follow. Exit nodes are used to pessimize your connectivity, by making it take a big detour. Why would we want p2p DERP traffic (which bypasses the exit node) to be pessimized as well?

**Comment by danderson** [comment]
As an example: I'm in the US near Seattle, and for my desktop I now decide I want to use an exit node in Singapore. Now, everything wanting to talk to my desktop has to haul all the way to Singapore and back, even if it's in the next house over?

**Comment by bradfitz** [comment]
Yeah, but it still might be the desirable trade-off (especially on mobile). If we need to keep our DERP home's TCP connection forever, and regular phone network chatter with an exit node means our exit node's DERP will likely always stay hot, I'd rather coalesce those two TCP connections down to one, even if it means slightly slower direct connection establishment (or higher latency in the hard-NAT-so-much-DERP case)

**Comment by danderson** [comment]
I disagree that this is a worthwhile tradeoff. Let's talk tomorrow?
